### PR TITLE
mark request as failed for runaway exceptions

### PR DESCRIPTION
**Pull request by SergKanz**

### CODE REVIEW

File: src/Microsoft.ApplicationInsights.AspNet/RequestTrackingMiddleware.cs [hunk 1]
```diff
@@ -32,31 +29,33 @@ public RequestTrackingMiddleware(RequestDelegate next, TelemetryClient client)
...
 
-                    telemetry.Timestamp = now;
-                    telemetry.Duration = sw.Elapsed;
-                    telemetry.ResponseCode = httpContext.Response.StatusCode.ToString();
-                    telemetry.Success = httpContext.Response.StatusCode < 400;
-                    telemetry.HttpMethod = httpContext.Request.Method;
-                    telemetry.Url = httpContext.Request.GetUri();
-                    telemetry.Context.GetInternalContext().SdkVersion = this.sdkVersion;
+                telemetry.Duration = sw.Elapsed;
+                telemetry.ResponseCode = httpContext.Response.StatusCode.ToString();
```
Comment (abaranch) [inline src/Microsoft.ApplicationInsights.AspNet/RequestTrackingMiddleware.cs]: So it still will be 200. Are we planning to di smth with that?..

**Comment by abaranch** [comment]
Approved

File: src/Microsoft.ApplicationInsights.AspNet/RequestTrackingMiddleware.cs [hunk 2]
```diff
@@ -32,31 +29,33 @@ public RequestTrackingMiddleware(RequestDelegate next, TelemetryClient client)
...
-            var now = DateTimeOffset.UtcNow;
+            bool requestFailed = false;
 
             try
             {
                 await this.next.Invoke(httpContext);
             }
+            catch (Exception)
+            {
+                requestFailed = true;
```
Comment (upendras) [inline src/Microsoft.ApplicationInsights.AspNet/RequestTrackingMiddleware.cs]: should we rethrow this exception?

Comment (SergKanz) [inline src/Microsoft.ApplicationInsights.AspNet/RequestTrackingMiddleware.cs]: Great catch! Thanks!

**Pull request merged**